From: John Hubbard <jhubbard@nvidia.com>
Subject: selftests/mm: mseal, self_elf: fix missing __NR_mseal
Date: Fri, 7 Jun 2024 19:10:19 -0700

Pach series "cleanups, fixes, and progress towards avoiding "make
headers"".

Eventually, once the build succeeds on a sufficiently old distro, the idea
is to delete $(KHDR_INCLUDES) from the selftests/mm build, and then after
that, from selftests/lib.mk and all of the other selftest builds.

For now, this series merely achieves a clean build of selftests/mm on a
not-so-old distro: Ubuntu 23.04:

1. Add __NR_mseal.

2. Add fs.h, taken as usual from a snapshot of
   ./usr/include/linux/fs.h after running "make headers".  This is how we
   have agreed to do this sort of thing, see [1].

3. Add a few selected prctl.h values that the ksm and mdwe tests require.

[1] commit e076eaca5906 ("selftests: break the dependency upon local
header files")


This patch (of 5):

The selftests/mm build isn't exactly "broken", according to the current
documentation, which still claims that one must run "make headers", before
building the kselftests.  However, according to the new plan to get rid of
that requirement [1], they are future-broken: attempting to build
selftests/mm *without* first running "make headers" will fail due to not
finding __NR_mseal.

Therefore, add __NR_mseal, to a new mseal_helpers.h file.  That file is
small right now, but subsequent patches will add a lot more content to it.

[1] commit e076eaca5906 ("selftests: break the dependency upon local
header files")

Link: https://lkml.kernel.org/r/20240608021023.176027-1-jhubbard@nvidia.com
Link: https://lkml.kernel.org/r/20240608021023.176027-2-jhubbard@nvidia.com
Fixes: 4926c7a52de7 ("selftest mm/mseal memory sealing")
Signed-off-by: John Hubbard <jhubbard@nvidia.com>
Reviewed-by: Jeff Xu <jeffxu@chromium.org>
Tested-by: Jeff Xu <jeffxu@chromium.org>
Cc: Andrei Vagin <avagin@google.com>
Cc: Axel Rasmussen <axelrasmussen@google.com>
Cc: Christian Brauner <brauner@kernel.org>
Cc: David Hildenbrand <david@redhat.com>
Cc: Kees Cook <kees@kernel.org>
Cc: Kent Overstreet <kent.overstreet@linux.dev>
Cc: Liam R. Howlett <Liam.Howlett@oracle.com>
Cc: Muhammad Usama Anjum <usama.anjum@collabora.com>
Cc: Peter Xu <peterx@redhat.com>
Cc: Rich Felker <dalias@libc.org>
Cc: Shuah Khan <shuah@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 tools/testing/selftests/mm/mseal_helpers.h |    5 +++++
 tools/testing/selftests/mm/mseal_test.c    |    1 +
 tools/testing/selftests/mm/seal_elf.c      |    1 +
 3 files changed, 7 insertions(+)

--- /dev/null
+++ a/tools/testing/selftests/mm/mseal_helpers.h
@@ -0,0 +1,5 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+
+#ifndef __NR_mseal
+#define __NR_mseal 462
+#endif
--- a/tools/testing/selftests/mm/mseal_test.c~selftests-mm-mseal-self_elf-fix-missing-__nr_mseal
+++ a/tools/testing/selftests/mm/mseal_test.c
@@ -17,6 +17,7 @@
 #include <sys/ioctl.h>
 #include <sys/vfs.h>
 #include <sys/stat.h>
+#include "mseal_helpers.h"
 
 /*
  * need those definition for manually build using gcc.
--- a/tools/testing/selftests/mm/seal_elf.c~selftests-mm-mseal-self_elf-fix-missing-__nr_mseal
+++ a/tools/testing/selftests/mm/seal_elf.c
@@ -16,6 +16,7 @@
 #include <sys/ioctl.h>
 #include <sys/vfs.h>
 #include <sys/stat.h>
+#include "mseal_helpers.h"
 
 /*
  * need those definition for manually build using gcc.
_
